plugins/modules/win_timezone.ps1

```diff
@@ -3,71 +3,75 @@
+}
 
-$timezone = Get-AnsibleParam -obj $params -name "timezone" -type "str" -failifempty $true
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
 
-$result = @{
-    changed = $false
-    previous_timezone = $timezone
-    timezone = $timezone
-}
+$timezone = $module.Params.timezone
 
-Try {
-    # Get the current timezone set
-    $result.previous_timezone = $(tzutil.exe /g)
-    If ($LASTEXITCODE -ne 0) {
-        Throw "An error occurred when getting the current machine's timezone setting."
-    }
+$module.Result.previous_timezone = $null
+$module.Result.timezone = $timezone
 
-    if ( $result.previous_timezone -eq $timezone ) {
-        Exit-Json $result "Timezone '$timezone' is already set on this machine"
-    }
-    Else {
-        # Check that timezone is listed as an available timezone to the machine
-        $tzList = $(tzutil.exe /l).ToLower()
-        If ($LASTEXITCODE -ne 0) {
-            Throw "An error occurred when listing the available timezones."
-        }
+Function Invoke-TzUtil {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory)]
+        [object]
+        $Module,
 
-        $tzExists = $tzList.Contains(($timezone -Replace '_dstoff').ToLower())
-        if (-not $tzExists) {
-            Fail-Json $result "The specified timezone: $timezone isn't supported on the machine."
-        }
+        [Parameter(Mandatory)]
+        [string]
+        $Action,
 
-        if ($check_mode) {
-            $result.changed = $true
-        }
-        else {
-            tzutil.exe /s "$timezone"
-            if ($LASTEXITCODE -ne 0) {
-                Throw "An error occurred when setting the specified timezone with tzutil."
-            }
-
-            $new_timezone = $(tzutil.exe /g)
-            if ($LASTEXITCODE -ne 0) {
-                Throw "An error occurred when getting the current machine's timezone setting."
-            }
-
-            if ($timezone -eq $new_timezone) {
-                $result.changed = $true
-            }
-        }
+        [Parameter(ValueFromRemainingArguments)]
+        [string[]]
+        $ArgumentList
+    )
 
-        if ($diff_support) {
-            $result.diff = @{
-                before = "$($result.previous_timezone)`n"
-                after = "$timezone`n"
-            }
-        }
+    $stdout = $null
+    $stderr = . { tzutil.exe @ArgumentList | Set-Variable -Name stdout } 2>&1 | ForEach-Object ToString
+
+    if ($LASTEXITCODE) {
+        $Module.Result.stdout = $stdout -join "`n"
+        $Module.Result.stderr = $stderr -join "`n"
+        $Module.Result.rc = $LASTEXITCODE
+        $Module.FailJson("An error occurred when $Action.")
     }
+
+    $stdout
+}
+
+# Get the current timezone set
+$previousTz = Invoke-TzUtil /g -Module $module -Action "getting the current machine's timezone setting"
+$module.Result.previous_timezone = $previousTz
+
+if ($module.DiffMode) {
+    $module.Diff.before = "$previousTz`n"
+    $module.Diff.after = "$timezone`n"
```

Comment:
Instead of `diff` being part of the result, it's stored under a specific property `$module.Diff`. The structure is the same as before where we typically return a `before` and `after` with the changes involved.